### PR TITLE
Comments, docstrings, extraction of soccer documents.

### PR DESCRIPTION
**Pull request by EnayatUllah**
* Comments added to functions and wherever required.
* Docstrings added to functions.
* Extracted documents with ids in soccer_ids, and saved them as separate files in the provided output directory.

### CODE REVIEW

File: get_soccer_players_articles.py [hunk 1]
```diff
@@ -9,26 +9,44 @@
 
 DEBUG = True
 
-
+''' Returns a list with wiki ids from the input file filein'''
```
Comment (marfox) [inline get_soccer_players_articles.py]: function docstrings should be placed **inside** the function body

File: get_soccer_players_articles.py [hunk 2]
```diff
@@ -9,26 +9,44 @@
...
+'''Prints out article present in path corpus directory if its ID is present in "soccer_ids" list'''
 def extract_soccer_articles(soccer_ids, corpus_dir, output_dir):
+    # Iterating over the files and sub-directories from corpus directory using os.walk function
     for path, subdirs, files in os.walk(corpus_dir):
+        # Iterating over all the files
         for name in files:
             f = os.path.join(path, name)
-            with open(f) as i:
+            # Opening a single file f
+            with open(f,'r') as i:
```
Comment (marfox) [inline get_soccer_players_articles.py]: the `r` parameter is not needed, as it is the default one

File: get_soccer_players_articles.py [hunk 3]
```diff
@@ -9,26 +9,44 @@
...
                 content = ''.join(i.readlines())
-            match = re.search('id="([^"]+)"', content)
-            current_id = match.group(1)
-            if DEBUG:
-                print "File = [%s] - Wiki ID = [%s]" % (f, current_id)
-            if current_id in soccer_ids:
-                shutil.copy(f, output_dir)
+            # Finding ALL the doc ids in content using a simple regex.
```
Comment (marfox) [inline get_soccer_players_articles.py]: Why are you changing the intended behavior here?
This is not needed, and also quite inefficient compared to the old one.
You are adding:
1.  a `for` loop
2.  a regex for each document
3. a filehandle for each document

Comment (EnayatUllah) [inline get_soccer_players_articles.py]: Yes, but the previous implementation was copying the whole wiki file( containing multiple articles). So there were articles not at all related to soccer( and thus their id's not in soccer_ids). So, aren't we supposed to extract a sub-corpus wherein each document's id is within the list of soccer_ids?

Comment (marfox) [inline get_soccer_players_articles.py]: No, see [this comment](../15#issuecomment-78995989).
In particular, [this line](../../blob/master/extract_verbs.sh#L111) of `extract_verbs.sh` is crucial to understand why.
``

Comment (EnayatUllah) [inline get_soccer_players_articles.py]: Both the links are not opening. 404 for both of them.

**Comment by marfox** [comment]
They are fixed now, sorry.

File: get_soccer_players_articles.py (hunk 3 above)
Comment (EnayatUllah) [inline get_soccer_players_articles.py]: @marfox Yes, I did see the csplit function. But there is no explicit check for soocer_id's there. So a wiki doc in which out of ,say n articles, only 1 is related to soccer, is copied by the get_soocer_players_articles script. And the csplit in extract.sh just segregates it into multiple files. And so the articles totally unrelated to soccer end up in the final docs.( This is the result I obtained when I did run the code).

This is what I recollect from the two scripts. I perhaps may be fundamentally wrong regarding it. Please correct me then. Thanks!

Comment (marfox) [inline get_soccer_players_articles.py]: The `get_soccer_players.py` script takes as input the corpus dir 
**after** the `csplit` run
So, **each** file corresponds to a **single** article.
You probably ran that script with a **non-split** corpus as input.

Comment (EnayatUllah) [inline get_soccer_players_articles.py]: Okay! Yes I did! Thanks for the clarification! So, this issue just involves the addition of comments and docstrings with no major modification in code, right?

Comment (marfox) [inline get_soccer_players_articles.py]: Yes, if you read the open tasks instructions, we would not be discussing 
about this now.

**Pull request closed without merging**